<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find turtlebot3_navigation)/maps/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="real_x" default="1.0"/>
  <arg name="real_y" default="2.0"/>
  <arg name="fake_x" default="1.2"/>
  <arg name="fake_y" default="2"/>
  

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL -->
  <include file="$(find multi_robot_nav)/launch/single_movebase_files/amcl.launch">
    <arg name="initial_pose_x" value="$(arg fake_x)"/>
    <arg name="initial_pose_y" value="$(arg fake_y)"/>
  </include>

  <!-- move_base -->
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />

  <!-- move_base node -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find multi_robot_nav)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find multi_robot_nav)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find multi_robot_nav)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find multi_robot_nav)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>

  <!-- rviz -->
    <!-- args="-d $(find multi_robot_nav)/rviz/turtlebot3_navigation.rviz"/> -->
  <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find multi_robot_nav)/rviz/myconfig.rviz"/>
</launch>
